core/Pos/Binary/Core/Types.hs

```diff
@@ -5,54 +5,53 @@ import           Universum
 import           Data.Time.Units            (Millisecond)
 import           Serokell.Data.Memory.Units (Byte)
 
-import           Pos.Binary.Class           (Bi (..), Cons (..), Field (..), Size (..),
-                                             UnsignedVarInt (..), deriveSimpleBi, label,
-                                             labelP, labelS, putField, putWord8)
-import qualified Pos.Binary.Core.Coin       as BinCoin
+import           Pos.Binary.Class           (Bi (..), Cons (..), Field (..), deriveSimpleBi)
 import           Pos.Binary.Core.Fee        ()
 import           Pos.Binary.Core.Script     ()
 import           Pos.Binary.Core.Version    ()
+import qualified Pos.Binary.Core.Coin       as BinCoin
 import qualified Pos.Core.Fee               as T
 import qualified Pos.Core.Types             as T
 import qualified Pos.Data.Attributes        as A
-import           Pos.Util.Util              (eitherToFail)
 
 -- kind of boilerplate, but anyway that's what it was made for --
 -- verbosity and clarity
 
 instance Bi T.Timestamp where
-    sizeNPut = labelS "Timestamp" $ putField toInteger
-    get = label "Timestamp" $ fromInteger <$> get
+  encode (T.Timestamp ms) = encode . toInteger $ ms
+  decode = T.Timestamp . fromIntegral <$> decode @Integer
 
 instance Bi T.TimeDiff where
-    sizeNPut = labelS "TimeDiff" $ putField toInteger
-    get = label "TimeDiff" $ fromInteger <$> get
+    encode = encode . toInteger
+    decode = fromInteger <$> decode
 
 instance Bi T.EpochIndex where
-    sizeNPut = labelS "EpochIndex" $ putField (UnsignedVarInt . T.getEpochIndex)
-    get = label "EpochIndex" $ T.EpochIndex . getUnsignedVarInt <$> get
+  encode (T.EpochIndex epoch) = encode epoch
+  decode = T.EpochIndex <$> decode
 
 instance Bi (A.Attributes ()) where
-    size = VarSize $ A.sizeAttributes (\() -> [])
-    get = label "Attributes" $
-        A.getAttributes (\_ () -> Nothing) (Just (128 * 1024 * 1024)) ()
-    put = labelP "Attributes" . A.putAttributes (\() -> [])
+  encode = A.encodeAttributes []
+  decode = A.decodeAttributes () $ \_ _ _ -> Nothing
 
 instance Bi T.Coin where
-    size = VarSize BinCoin.size
-    put = labelP "Coin" . mapM_ putWord8 . BinCoin.encode
-    get = label "Coin" $ BinCoin.decode
+  encode = encode . BinCoin.encode
```

Comment:
If you use [decimal-fraction encoding](https://tools.ietf.org/html/rfc7049#section-2.4.3), the difference will be around 2× (I checked), which is acceptable. Otherwise, common values like one ADA will be encoded as 1000000 coins = `1A 000F4240` = 5 bytes, which is kinda worse and I'd rather prefer decimal-fraction encoding here.